src/common/cockpitwebresponse.c

```diff
@@ -234,7 +234,11 @@ cockpit_web_response_new (GIOStream *io,
   self->url_root = NULL;
   self->full_path = g_strdup (path);
   self->path = self->full_path;
-  cockpit_web_response_set_method (self, method);
+
+  g_assert (method != NULL);
+  if (!g_str_equal (method, "GET") && !g_str_equal (method, "HEAD"))
+    g_warning("Method %s is not supported", method);
```

Comment:
I don't like this.  Someone trying a random method on a public-listening webserver shouldn't end up in the journal... just get rid of this.

Reply:
Ack, dropped. I also flipped the GET/HEAD check as you suggested above, so that `PATCH` error responses now have a body.